sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/KeyCredentialPolicy.java

```diff
@@ -18,6 +20,7 @@
  * <p>Requests sent with this pipeline policy are required to use {@code HTTPS}. If the request isn't using {@code HTTPS}
  * an exception will be thrown to prevent leaking the key.</p>
  */
+@Metadata(properties = MetadataProperties.IMMUTABLE)
```

Comment:
Do we want to have `IMMUTABLE` annotation on policies? Should we reserve the fluent vs immutable annotations for models?

Reply:
Let's leave it there as we don't actively use these annotations anywhere.